apps/sasquatch/src/projectDependency/java/com/microsoft/appcenter/sasquatch/activities/StorageActivity.java

```diff
@@ -0,0 +1,141 @@
+        mStorageTypeSpinner.setAdapter(typeAdapter);
+        mStorageTypeSpinner.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
+
+            @Override
+            public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
+                updateStorageType(position);
+            }
+
+            @Override
+            public void onNothingSelected(AdapterView<?> parent) {
+            }
+        });
+
+        listView = findViewById(R.id.list);
+        listView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
+            @Override
+            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
+            }
+        });
+
+
+        /* TODO remove reflection once Storage published to jCenter. */
+        try {
+            Class<?> storage = Class.forName("com.microsoft.appcenter.storage.Storage");
+            createDocument(storage);
+            readDocument(storage);
+            deleteDocument(storage);
+        } catch (Exception e) {
+            Log.e(LOG_TAG, "Storage.Module call failed", e);
+        }
+    }
+
+    private void updateStorageType(int position) {
+        switch (position) {
+            case 0:
+                listView.setAdapter(new ArrayAdapter<String>(this, android.R.layout.simple_list_item_1, appDocumentList));
+                break;
+            case 1:
+                if (AuthenticationProviderActivity.userID != null) {
+                    listView.setAdapter(new ArrayAdapter<String>(this, android.R.layout.simple_list_item_1, userDocumentList));
+                } else {
+                    String[] signInReminder = {"Please Sign In Identity First Before Get User Document!"};
+                    listView.setAdapter(new ArrayAdapter<String>(this, android.R.layout.simple_list_item_1, signInReminder));
+                }
+                break;
+        }
+    }
+
+    private void listDocuments(Class<?> storage, String partition) throws IllegalAccessException, InvocationTargetException, NoSuchMethodException {
+        Object future = storage.getMethod("list", String.class, Class.class).invoke(null, partition, TestDocument.class);
+        ((AppCenterFuture<PaginatedDocuments<TestDocument>>) future).thenAccept(new AppCenterConsumer<PaginatedDocuments<TestDocument>>() {
+            @Override
+            public void accept(PaginatedDocuments<TestDocument> documents) {
+                int listSize = documents.getCurrentPage().getItems().size();
+                for (int i = 0; i < listSize; i++) {
+                    appDocumentList.add(documents.getCurrentPage().getItems().get(i).getId());
+                }
+            }
+
+        });
+    }
+
+    private void deleteDocument(Class<?> storage) throws IllegalAccessException, InvocationTargetException, NoSuchMethodException {
```

Comment:
Could not find a way to remove a document in the UI, we need a control

Reply:
Add it in new commit.